tests/foreman/cli/test_host.py

```diff
@@ -1847,8 +1838,20 @@ def _make_activation_key(self, add_subscription=False):
             )
         return activation_key
 
-    def _host_subscription_register(self):
+    def _host_subscription_register(self, request):
```

Comment:
Couldn't the new teardown method do this and not require tests to add a `request` argument?

Reply:
This do not work when the host is registered only from hammer. The record will stay in satellite. If new machine with the same name arrives there will be unwanted behaviour, example: organization mismatch so it will not be registered and test will fail. 

```
Host.subscription_register ->  hammer host subscription register <params>
```

So it has to be cleaned from host and from satellite. Maybe we can introduce smth similar we did for host. 
I am open to ideas.

Reply:
I have to say I didn't test it. If you want me to I will.